### ISSUE TITLE
url_handlers vs scope_extensions

### ISSUE THREAD

**Issue by jaroslawdabrowski** [MicrosoftDocs/edge-developer#2155]
Hi Edge Devs,

can you please tell me which property is now supported? url_handlers or scope_extensions? None works for me. We have a multi tenant web application for which we would like develop mobile apps using pwa. Of course we would like to have only one app, not for every tenant. We have the root domain that is being used as a starting point and it can direct users to their instances. This works fine for ios using allowedOrigins, works fine from android using additionalTrustedOrigins, but I cannot make it work for windows. There is no such configuration option for windows and I get the address bar for my domains outside the root domain. I was trying to use url_handlers and scope_extensions in my manifest, but nono of that worked out. Can you please help me with this?

Thanks in advance for your help.

[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d7f49d04-dd9e-9d95-6bc3-f158e4eacada
* Version Independent ID: 5e3a5695-0978-7051-45d1-623dbb757f22
* Content: [Publish a Progressive Web App to the Microsoft Store - Microsoft Edge Development](https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/microsoft-store)
* Content Source: [microsoft-edge/progressive-web-apps-chromium/how-to/microsoft-store.md](https://github.com/MicrosoftDocs/edge-developer/blob/main/microsoft-edge/progressive-web-apps-chromium/how-to/microsoft-store.md)
* Product: **microsoft-edge**
* Technology: **pwa**
* GitHub Login: @MSEdgeTeam
* Microsoft Alias: **msedgedevrel**

**Comment by diekus** [comment]
Hola @jaroslawdabrowski,

Here's some light on what is going on with `url_handlers` and `scope_extensions`. At the moment, `url_handlers` as a manifest field is not being considered nor worked on in Chromium. On the Edge side of things, we have a browser behaviour that is launching very soon that pretty much does the same thing as handling links for a specific domain. When you install a PWA on Windows, this PWA will automatically become the default link (or URL, if you wish) handler for the scope. For example, once I install Youtube as a PWA on Windows, if I click on `www.youtube.com/some-video-address` it will open in the Youtube PWA instead of in the browser. The user can control this behaviour from the app details page in `edge://apps`. There is a toggle that allows them to turn on/off this behaviour.

Regarding `scope_extensions`, this *is* actively being worked on atm and being led by the MS Edge team for Chromium. It is not ready, but when it lands on stable it will allow developers to expand the scope defined by the PWA, which could be used, for example, to include all TLD into the scope associated with a site.

But in summary, you are correct. `url_handlers` and `scope_extensions` currently do not work. One replaced by a modern browser behaviour, the other one still in development. Any other question I can help you with I am more than happy to reply.

**Comment by jaroslawdabrowski** [comment]
Cześć Diego!

Thank you for your response, it is very helpful. Do you have any idea when scope_extensions will be supported? Any chance it will be available until end of this year? I am especially interested in edge, as we want to mobilize our app using PWABuilder, desktop windows app is using the edge windows and there we are facing an issue with our multi tenant architecture. Scope extensions would be very helpful. Thank you in advance for your response.

**Comment by captainbrosset** [comment]
Thanks Diego.

We had added a note at the beginning of https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/handle-urls to that regard. It might actually be even better if we removed that page entirely. There's not much value in documenting `url_handlers` if it's going away.

Or we could keep it to document the customizable link handling behavior on Windows. And then, once `scope_extensions` is ready, add that to the same page.
